### PR TITLE
remove cloneType

### PR DESCRIPTION
**Pull request by sequencer**

### CODE REVIEW

**Comment by jiegec** [comment]
@aswaterman Any chances this pr can be merged? I am using berkeley-hardfloat with Chisel 3.5.0-RC2, which forbids the usage of cloneType.

**Comment by jackkoenig** [comment]
It's probably worth also updating the `build.sbt` so that this change can be tested

**Comment by aswaterman** [comment]
@jiegec can you or @sequencer follow @jackkoenig’s advice? I’m OK with merging the PR subsequently.

**Comment by sequencer** [comment]
Ok I’ll do tonight.

File: build.sbt [hunk 1]
```diff
@@ -13,12 +13,8 @@ resolvers ++= Seq(
...
 )
 
-val defaultVersions = Map("chisel3" -> "3.3-SNAPSHOT")
-
-// Provide a managed dependency on chisel if -DchiselVersion="" issupplied on the command line.
-libraryDependencies ++= (Seq("chisel3").map {
-  dep: String => "edu.berkeley.cs" %% dep % sys.props.getOrElse(dep + "Version", defaultVersions(dep))
-})
+addCompilerPlugin("edu.berkeley.cs" % "chisel3-plugin" % "3.5.0-SNAPSHOT" cross CrossVersion.full)
+libraryDependencies += "edu.berkeley.cs" %% "chisel3" % "3.5.0-SNAPSHOT"
```
Comment (jackkoenig) [inline build.sbt:17]:
```suggestion
val chiselVersion = "3.5.0-RC2"
addCompilerPlugin("edu.berkeley.cs" % "chisel3-plugin" % chiselVersion cross CrossVersion.full)
libraryDependencies += "edu.berkeley.cs" %% "chisel3" % chiselVersion
```
I'd use 3.5.0-RC2, I also think we should move rocket-chip this direction

Comment (sequencer) [reply build.sbt]: Sure.

Comment (sequencer) [reply build.sbt]: I'd like to keep `"edu.berkeley.cs" % "chisel3-plugin" % "3.5.0-RC2" cross CrossVersion.full`, I think if adding scala-steward in the future, it cannot regex this pattern.

**Comment by sequencer** [comment]
test successfully on my own PC.(I think I need PR a CI)

File: build.sc [hunk 2]
```diff
@@ -7,7 +7,7 @@ import coursier.maven.MavenRepository
 // Please retain it.
 // Provide a managed dependency on X if -DXVersion="" is supplied on the command line.
 val defaultVersions = Map(
-  "chisel3" -> "3.3-SNAPSHOT",
+  "chisel3" -> "3.5-SNAPSHOT",
```
Comment (jackkoenig) [inline build.sc:10]:
```suggestion
  "chisel3" -> "3.5.0-RC2",
```

**Review by jackkoenig — APPROVED** [review]
Tests pass locally for me as well, but all I ran was `make`, I did not test the mill build.

File: build.sc (hunk 2 above)
Comment (sequencer) [reply build.sc:10]: nice catch, I forgot to add plugin as well.

**Comment by sequencer** [comment]
ready for merge

**Comment by jiegec** [comment]
ping @aswaterman

**Pull request merged**